web/src/narrow.js

```diff
@@ -957,11 +957,11 @@ export function to_compose_target() {
     };
 
     if (compose_state.get_message_type() === "stream") {
-        const stream_name = compose_state.stream_name();
-        const stream_id = stream_data.get_stream_id(stream_name);
+        const stream_id = compose_state.stream_id();
         if (!stream_id) {
             return;
         }
+        const stream_name = stream_data.get_sub_by_id(stream_id).name;
```

Comment:
I believe the idea was to remove situations where we get id from name, and instead default to id and sometimes get name from id. 

It's better to start with the id because it's a more robust identifier (will never change for a stream object), so if we have to sometimes get name from id or sometimes get id from name it's better to start with id.

Reply:
Maybe we can use `compose_state.stream_name()` here? It uses the id as well.

Here we do `const stream_name = stream_data.get_sub_by_id(stream_id).name;` which uses `stream_data.stream_info` for lookup.

and in `stream_name` it does `sub_store.maybe_get_stream_name(stream_id)` which eventually does `subs_by_stream_id.get(stream_id)`

It's a bit strange that this data is stored in two different places, but the latter maps to `StreamSubscription` and the first is a much simpler map.

I'm not sure if there is much benefit to doing one or the other.

Reply:
The main idea behind my comment is simple - let's utilize functions like `compose_state.stream_name()` to retrieve the `stream_name` instead of writing custom logic. By doing so, we can reduce the likelihood of bugs and improve maintainability.

If we are unable to use those functions and we're aware of the limitations, that's okay, I suppose.

Reply:
You have much more data and context about those functions, so I assume you have chosen the best option and are aware of it ;))

Reply:
No I think this is a great point!

I don't always chose the best option or think about everything-- there's a lot to think about in each PR (especially this one) and I was mostly biasing for simplicity. If we simplify `stream_name` overall I think that would be good for another PR. 

I'm still not sure if using `compose_state.stream_name()` is better or worse here -- maybe I'll leave that to @timabbott to add input on :)

Reply:
I don't think this is an immediate problem. The medium-term plan is perhaps going to involve removing `compose_state.stream_name` entirely -- I would expect essentially all the APIs to just want to start from the stream ID and get data from there.